### ISSUE TITLE
iOS version is not pulling the tokenissuer field from PrivacyIDEA

### ISSUE THREAD

**Issue by mgryka** [freeotp/freeotp-ios#308]
PrivacyIDEA settings: 

- tokenissuer = companyname
- tokenlabel = <u> (tag for user)

_Behavior on Android freeOTP:_
User scans QR code.
New token appears:

- **companyname**
- user@companyname.example

_Behavior on iOS freeOTP:_
User scans QR code.
User asked for input for token issuer.
User asked to choose icon.
New token appears:

- **user input**
- user@companyname.example

Am I misconfiguring something? The Android app is as expected.

**Comment by justin-stephenson** [comment]
Hi,

The token label should also contain the account name, in the format `issuer:account`, see https://freeotp.github.io/qrcode.html for example.

That said we could maybe handle this better by checking for the `issuer` parameter, instead of only relying on the label.

**Comment by mgryka** [comment]
Thanks for the reply!

I'll root around in PrivacyIDEA but from the looks of it there isn't much I can do to change the way this information is conveyed in the QR code.

It isn't a dealbreaker by any means, I just want to make this as easy as possible on my end users (for my sake! haha).

**Comment by mgryka** [comment]
After using the tool in the link you provided and digging a little bit deeper into the PrivacyIDEA settings, it looks like I was able to hardcode it as part of the label. I tested on iOS and it works as expected!
![](https://github.com/freeotp/freeotp-ios/assets/104231072/d7457a32-862e-45d7-96ae-85aad29eb9ae)

I really appreciate you taking the time to help me work through this @justin-stephenson Thanks again!
